demonstrations/tutorial_classical_kernels.py

```diff
@@ -0,0 +1,738 @@
+# In this demo we will consider only this one example, but the arguments we
+# make and the code we use are also amenable to any kernel with the following
+# mild restrictions:
+#
+# #. Shift-invariance
+# #. Normalization :math:`k(0)=1`.
+# #. Smoothness (in the sense of a quickly decaying Fourier spectrum).
+#
+# Note that is a very large class of kernels!
+# And also an important one for practical applications.
+#
+# Fourier analysis of the Gaussian kernel
+# -------------------------------------------
+#
+# The first next step will be to find the Fourier spectrum of the Gaussian
+# kernel, which is an easy problem for classical computers.
+# Once we've found it, we'll build a QK that produces a finite Fourier series
+# approximation to that spectrum.
+#
+# Let's briefly recall that a Fourier series is the representation of a
+# periodic function using the sine and cosine functions.
+# Fourier analysis tells us that we can write any given periodic function as
+#
+# .. math:: f(x) = a_0 + \sum_n a_n\cos(n\omega_0x) + b_n\sin(n\omega_0x).
+#
+# For that, we only need to find the suitable base frequency :math:`\omega_0`
+# and coefficients :math:`a_0, a_1, \ldots, b_0, b_1,\ldots`.
+#
+# But the Gaussian kernel is an aperiodic function, whereas the Fourier series
+# only makes sense for periodic functions!
+#
+# *What can we do?!*
+#
+# We can cook up a periodic extension to the Gaussian kernel, for a given
+# period :math:`2L` (we take :math:`2\pi` as default):
+
+def Gauss_p(x, L=np.pi):
+    # Send x to x_mod in the period around 0
+    x_mod = np.mod(x+L, 2*L) - L
+
+    return gaussian_kernel(x_mod)
+
+###############################################################################
+# which we can now plot
+
+x_func = np.linspace(-10, 10, 123)
+y_func = [Gauss_p(x) for x in x_func]
+
+plt.plot(x_func, y_func)
+plt.xlabel("$\delta$")
+plt.suptitle("Periodic extension to the Gaussian kernel")
+plt.show();
+
+##################################################################################
+# In practice we'll be interested in cases where the period approaches
+# infinity.
+#
+# Next up, how does the Fourier spectrum of such an object look like?
+# We can find out using PennyLane's ``fourier`` module!
+
+from pennylane.fourier import coefficients
+
+##################################################################################
+# The function ``coefficients`` computes for us the coefficients of the Fourier
+# series up to a fixed term.
+# One tiny detail here: ``coefficients`` returns one complex number :math:`c_n`
+# for each frequency :math:`n`.
+# The real part corresponds to the :math:`a_n` coefficient, and the imaginary
+# part to the :math:`b_n` coefficient: :math:`c_n=a_n+ib_n`.
+# Because the Gaussian kernel is an even function, we know the imaginary part
+# of every coefficient will be zero, so :math:`c_n=a_n`.
+
+def fourier_p(d):
+    # we only take the first d coefficients [:d]
```

Comment:
```suggestion
    # We only take the first d coefficients [:d]
```

If it has a fullstop the first word is capitalised...

Reply:
correcto! That one slipped past me